core/src/main/java/io/grpc/internal/SerializingExecutor.java

```diff
@@ -87,7 +92,7 @@ private void schedule(@Nullable Runnable removable) {
             // to execute don't succeed and accidentally run a previous runnable.
             runQueue.remove(removable);
           }
-          running.set(false);
+          runState = STOPPED;
```

Comment:
You must use `runStateUpdater` for mutations. From the `AtomicIntegerFieldUpdater` documentation:
> it can guarantee atomicity only with respect to other invocations of compareAndSet and set on the same updater

Reply:
So is it error prone? Future coder may assign values on it directly without noticing the existence of the updater.

Reply:
Yeah, it will be error prone.  I suggested this as an ErrorProne warning.